accesscontroltool-bundle/src/main/java/biz/netcentric/cq/tools/actool/authorizableinstaller/impl/AuthorizableInstallerServiceImpl.java

```diff
@@ -226,11 +231,25 @@ private void installKeys(Map<String, Key> keys, String userId, ResourceResolver
         }
     }
 
-
     void setUserPassword(final AuthorizableConfigBean authorizableConfigBean,
-                             final User authorizableToInstall) throws RepositoryException, AuthorizableCreatorException {
+            final User authorizableToInstall, InstallationLogger installLog) throws RepositoryException, AuthorizableCreatorException {
+
+        String userId = authorizableToInstall.getID();
         String password = getPassword(authorizableConfigBean);
-        authorizableToInstall.changePassword(password);
+        Session sessionForUser = null;
+        try {
+            sessionForUser = repository.login(new SimpleCredentials(userId, password.toCharArray()));
```

Comment:
If it was possible to use the oak security provider (as in https://github.com/apache/jackrabbit-oak/blob/15e2befce204f0a3b7457777d912db06c93d235d/oak-core/src/main/java/org/apache/jackrabbit/oak/core/ContentRepositoryImpl.java#L161-L163) directly it would be more light-weight, to me it seems not easily possible though...

Reply:
Oak Core is internal API and subject to change, so the approach with JCR API is a good approach.